src/arrayMethodSort.js

```diff
@@ -1,12 +1,83 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const undefinedArr = [];
+
+    for (let i = 0; i < this.length; ) {
+      if (this[i] === null) {
+        undefinedArr.unshift(...this.splice(i, 1));
+      }
+
+      if (typeof this[i] === 'undefined') {
+        undefinedArr.push(...this.splice(i, 1));
+      } else {
+        this[i] = this[i].toString();
+        i++;
+      }
+    }
+
+    for (let j = 0; j < this.length; j++) {
+      for (let i = 0; i < this.length; i++) {
+        let result;
+
+        if (i + 1 < this.length) {
+          if (compareFunction) {
+            result = compareFunction(this[i], this[i + 1]);
+          }
+
+          if (!compareFunction) {
+            if (this[i] > this[i + 1]) {
+              result = 1;
+            }
+          }
```

Comment:
The condition `!compareFunction` is redundant because the `compareFunction` is already checked in the previous condition. Consider using an `else` statement instead.